src/main/java/com/dev/cinema/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
```

Comment:
itemId?
copy-paste is evil)))

Reply:
)))evil is laziness, but not copy-paste)
Done